### DISCUSSION TITLE
Access denied after a few hours

### DISCUSSION THREAD

**Discussion by cnxsoft** [shlinkio/discussions#47] [Help wanted] [answered]
#### How Shlink is set-up

* Shlink Version: 2.8.1
* PHP Version: 8.0
* How do you serve Shlink: Self-hosted nginx
* Database engine used: MariaDB

#### Summary

I'm migrated an existing Shlink database using url.cnx-software.com to a new server and a new domain (url2.cnx-software.com), and it's working fine but only for a time, usually a few hours. After that the URL will not resolve with the error similar to "unexpected error" showing in the web browser.

If I use the CLI application (sudo -u www-data php8.0 bin/cli api-key:list) it shows a mysql 1045 access denied issue (Sorry, I don't have the error message right now).

The way I "fix" the issue right now is to run the bin/install script, and it's working. But I'd expect it to fail again soon. I'll make sure to update the report with the exact error message once it happens.

**Comment by cnxsoft** [comment, marked as answer]
Here's the exact error I get in the web browser:
"An unexpected error occurred"

shlink.error is empty.

Example of command that fails:

`sudo -u www-data php8.0 bin/cli api-key:list

In AbstractMySQLDriver.php line 112:

  An exception occurred in driver: SQLSTATE[HY000] [1045] Access denied for u
  ser 'shlink'@'localhost' (using password: YES)


In Exception.php line 18:

  SQLSTATE[HY000] [1045] Access denied for user 'shlink'@'localhost' (using p
  assword: YES)


In PDOConnection.php line 39:

  SQLSTATE[HY000] [1045] Access denied for user 'shlink'@'localhost' (using p
  assword: YES)


api-key:list [-e|--enabled-only] [--enabledOnly]`

It's like shlink will forget the password after a while...

**Reply by acelaya** [reply]
All the custom configuration set during installation, is persisted in the `config/params/generated_config.php`.

Maybe there's something in the system deleting it.

**Reply by cnxsoft** [reply]
I know what it is now... It was my mistake sorry about that. I misunderstood how rsync works by default when syncing files between servers. With the options I used, it would overwrite the files if they are different, regardless of the time of the last modification...
